pints/_mcmc/_slice_rank_shrinking.py

```diff
@@ -0,0 +1,265 @@
+# -*- coding: utf-8 -*-
+#
+# Slice Sampling - Covariance Adaptive: Rank Shrinking
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+
+
+class SliceRankShrinkingMCMC(pints.SingleChainMCMC):
+    """
+    Implements Covariance-Adaptive Slice Sampling by Rank Shrinking,
+    as described in [1]. This is an adaptive multivariate method which
+    uses additional points, called "crumbs", and rejected proposals to
+    guide the selection of samples.
+
+    It generates samples by sampling uniformly from the volume underneath the
+    posterior (``f``). It does so by introducing an auxiliary variable (``y``)
+    and by definying a Markov chain.
+
+    Sampling follows:
+
+    1. Calculate the pdf (``f(x_0)``) of the current sample (``x_0``).
```

Comment:
Again. Please update all the equations. See other PRs for details

Reply:
Done